src/main/resources/plugin-settings.template.html

```diff
@@ -15,94 +15,147 @@
+			padding: 0 !important;
+		}
+
+		[data-plugin-style-id="kubernetes-plugin"] fieldset {
+			border:           1px solid #ddd;
+			padding:          20px;
+			margin-bottom:    20px;
+			border-radius:    3px;
+			background-color: transparent;
+		}
+
+		[data-plugin-style-id="kubernetes-plugin"] fieldset legend {
+			padding:          5px;
+			font-size:        0.875rem;
+			margin-bottom:    0px;
+			background-color: #fff;
+		}
+
+		[data-plugin-style-id="kubernetes-plugin"] .form-help-content {
+			color:         #666;
+			clear:         both;
+			font-size:     0.82rem;
+			font-style:    italic;
+			margin-top:    -15px;
+			padding-left:  2px;
+			margin-bottom: 10px;
+		}
+
+		[data-plugin-style-id="kubernetes-plugin"] .form-help-content code {
+			border:           0px;
+			padding:          2px 5px;
+			border-radius:    3px;
+			background-color: #eee;
+		}
+
+		[data-plugin-style-id="kubernetes-plugin"] .form-help-content .code {
+			padding:          16px;
+			overflow:         auto;
+			font-size:        85%;
+			line-height:      1.45;
+			border-radius:    3px;
+			background-color: #f6f8fa;
+		}
+	</style>
+
+	<div class="row">
+		<label>Go Server URL</label>
+		<input type="text" ng-model="go_server_url" ng-required="false"/>
+		<span class="form_error" ng-show="GOINPUTNAME[go_server_url].$error.server">{{GOINPUTNAME[go_server_url].$error.server}}</span>
+		<label class="form-help-content">
+			Server hostname must resolve in your container. Don't use <code>localhost</code> or <code>127.0.0.1</code>.
+			Defaults to GoCD Secure site URL if not specified.
+		</label>
+	</div>
+
+	<div class="row">
+		<label>Agent auto-register timeout (in minutes)<span class='asterix'>*</span></label>
+		<input type="text" ng-model="auto_register_timeout" ng-required="true"/>
+		<span class="form_error" ng-show="GOINPUTNAME[auto_register_timeout].$error.server">{{GOINPUTNAME[auto_register_timeout].$error.server}}</span>
+		<label class="form-help-content">Defaults to <code>10 minutes</code>.</label>
+	</div>
+
+	<div class="row">
+		<label>Maximum pending pods<span class='asterix'>*</span></label>
+		<input type="text" ng-model="pending_pods_count" ng-required="true"/>
+		<span class="form_error" ng-show="GOINPUTNAME[pending_pods_count].$error.server">{{GOINPUTNAME[pending_pods_count].$error.server}}</span>
+		<label class="form-help-content">Defaults to <code>10 pods</code>.</label>
+	</div>
+
+	<fieldset>
+		<legend>Cluster credentials</legend>
+		<div class="row">
+			<div class="columns small-3 medium-2 larger-2 no-padding">
+				<label>Authentication Strategy</label>
+			</div>
+			<div class="columns small-9 medium-10 larger-10 no-padding"
+				 ng-init="authentication_strategy = (authentication_strategy || 'OAUTH_TOKEN')">
+				<input type="radio" ng-model="authentication_strategy" value="OAUTH_TOKEN" id="oauth-token"
+					   ng-checked="(authentication_strategy == 'OAUTH_TOKEN')"/>
+				<label for="oauth-token">OAuth token</label>
```

Comment:
Should we provide a note about how to obtain this for a service account of the user's choice?

Reply:
Hmm, sure. We should remember to come back and edit the view.